### PR TITLE
Iterations to link interface

### PR DESCRIPTION
**Pull request by karmatosed**
These few changes bring some of the visual iterations explored in #26351 into experimentation. You can see this currently on navigation block for example by adding a link. 

**Before:**

image

**After:**

image

What changes:

- Removes the horizontal line break
- Reduces padding

[ edit of additional changes ]

- Grid variables
- Margin changes
- Small label styling

This doesn't address the icon, as that needs to be confirmed. I am creating it so people can explore what this feels like in an actual pull request.

### CODE REVIEW

File: packages/block-editor/src/components/link-control/style.scss
Comment (jasmussen) [inline packages/block-editor/src/components/link-control/style.scss:215]: While you're here, it might be cool to update these to use variables, such as $grid-unit-15 and $grid-unit-20.

**Review by jasmussen — APPROVED** [review]
Simple and sweet, we need more PRs like these. It seems a good improvement. Before:

before

After:

interface

I left a comment in case you'd like to do a few more tweaks while you're in there, some of those properties feel a bit arbitrary. Like in the following, the margin below the highlighted item doesn't feel "griddy" (12px grid ideally but 8px in absence of that):

Screenshot 2020-10-29 at 09 50 20

Also, this "recently updated" feels weird. I'd make it 13px semibold font to serve as a subheading, and then be flusn on the left side with the input field:

Screenshot 2020-10-29 at 09 50 26

Those are all totally fine to consider separately if you prefer.

**Comment by karmatosed** [comment]
> I left a comment in case you'd like to do a few more tweaks while you're in there

Love that idea!

I am going to push each one to break them apart. Then I will collect them all in the issue so there's a record of the changes. I have made a variable as there wasn't one for the font size, I think having one could be useful. I could just set it to 13px though if not. Here is the result:

image

**Comment by karmatosed** [comment]
@jasmussen I just made a commit to this PR that I would love feedback on, that is regarding the removal of gradients on search results. Here is a before and after:

Before:

results


After:

image

I can see what use they are for longer results, but as they interfer with the hover, I am wondering if we could remove them? Or perhaps remove them on hover? I am interested in feedback on this.

**Comment by jasmussen** [comment]
From https://github.com/WordPress/gutenberg/pull/26551#issuecomment-719031153, yeah I think we should set that to 13px.

Also looking at it, removing the left indentation did not work as well as I'd hoped. I think the indentation of the page items below are probably the source of that problem, but in order to keep this PR small we should probably just add back that subheading indentation and look at the indentation overall as a separate task.

**Comment by karmatosed** [comment]
> look at the indentation overall as a separate task.

Sure, that's something I'm happy to explore later on as think a few areas could benefit from analysis over whether they indent or not and having some tightening up around that.

I just added a commit that brings in some more grid variables - great to align a few more while there. It also reverts the indent.

Here is what this now looks like:

image

**Comment by jasmussen** [comment]
Yep, that's good. Also good to remove that new variable from the stylesheet, they are worth adding occasionally but we should be careful adding too many, as they are meant to be used widely.

**Comment by karmatosed** [comment]
@jasmussen thanks, did a clean up. I have a few more things to add to this PR before merging too. I'll work on the icon in next pass as have that file now 🙌🏻

**Comment by karmatosed** [comment]
I have added the updates icon. Props to @jameskoster for help finding the location there.

**Before**

2020-11-06 at 18 44


**After**

after

**Comment by karmatosed** [comment]
My feelings are now that this PR I think is ready to get a final review then commit. Next steps will be to move into getting development on the other steps in issue.

**Comment by karmatosed** [comment]
One more additional icon change is to bring in the '+' with updated styling.

**Before**

image


**After**

image

## Further clean-up
This does open up more cleanup potential where the url icon isn't aligning with the +. Here you can see that quite clearly:

image

## Feedback
I didn't bring in the background as the adding blocks have, however this does help visually. There could be an argument for that here, so I am going to put on the feedback label again for some thoughts before moving to commit. 

## Next steps
This PR is ready once these icons have feedback to move to code review and potentially commit.

**Comment by jasmussen** [comment]
When trying to compile this I'm getting a "Error: Undefined variable. ... font-size: $helptext-font-size;`

**Comment by karmatosed** [comment]
I am just trying to get tests to pass but the variable has been updated and some iterations done to icon placement. After that as this had a code review, I will look to head to commit. Thanks everyone for the help in getting here.

File: packages/icons/src/library/keyboard-return.js
Comment (ellatrix) [inline packages/icons/src/library/keyboard-return.js:8]:
```suggestion
		<Path d="M6.734 16.106l2.176-2.38-1.093-1.028-3.846 4.158 3.846 4.157 1.093-1.027-2.176-2.38h2.811c1.125 0 2.25.03 3.374 0 1.428-.001 3.362-.25 4.963-1.277 1.66-1.065 2.868-2.906 2.868-5.859 0-2.479-1.327-4.896-3.65-5.93-1.82-.813-3.044-.8-4.806-.788l-.567.002v1.5c.184 0 .368 0 .553-.002 1.82-.007 2.704-.014 4.21.657 1.854.827 2.76 2.657 2.76 4.561 0 2.472-.973 3.824-2.178 4.596-1.258.807-2.864 1.04-4.163 1.04h-.02c-1.115.03-2.229 0-3.344 0H6.734z" />
```

This should fix a lint error.

Comment (karmatosed) [reply packages/icons/src/library/keyboard-return.js]: Thanks @ellatrix just pushed a hopeful resolution for that and also ran snapshot.

**Comment by karmatosed** [comment]
Looks like tests have passed so going to commit this one. Thank you everyone.

**Pull request merged**
